bgpd/bgp_mplsvpn.c

```diff
@@ -1528,10 +1747,20 @@ void vpn_leak_from_vrf_update(struct bgp *to_bgp,	     /* to */
 		nexthop_self_flag = 1;
 	}
 
-	label_val = from_bgp->vpn_policy[afi].tovpn_label;
-	if (label_val == MPLS_LABEL_NONE) {
+	label_val = vpn_leak_from_vrf_get_label_for_path_vrf(
+		afi, safi, path_vrf, from_bgp, to_bgp);
+
+	/* if label allocation or reachability checks tasks are in progress
+	 * then vpn_leak_from_vrf_update() will be called later
+	 */
+	if (label_val == MPLS_INVALID_LABEL) {
+		if (debug)
+			zlog_debug("%s: %s skipping: label per nexthop needed",
+				   __func__, from_bgp->name_pretty);
```

Comment:
if `from_bgp->vpn_policy[afi].tovpn_label` equals `MPLS_INVALID_LABEL` does not mean per nexthop is needed

use a new dedicated `MPLS_XXXX` value to notice that nexthop label has not been resolved

Reply:
after discussion, to ensure that the returned value is limited to the per nexthop label allocation case, check that the address family is correctly configured.